### PR TITLE
Upgrade lifecycle method & Add renderData, renderAllData

### PR DESCRIPTION
**Pull request by jyash97**
@metagrover Is this correct way to use `renderData` ? How we are planning to use `onData` , I would add its support in this.

### CODE REVIEW

File: packages/web/src/components/result/ReactiveList.js [hunk 1]
```diff
@@ -118,48 +118,48 @@ class ReactiveList extends Component {
...
 				options.sort = [{
-					[nextProps.dataField]: {
-						order: nextProps.sortBy,
+					[this.props.dataField]: {
+						order: this.props.sortBy,
 					},
 				}];
 			}
-			this.props.setQueryOptions(this.props.componentId, options, true);
+			prevProps.setQueryOptions(prevProps.componentId, options, true);
```
Comment (metagrover) [inline packages/web/src/components/result/ReactiveList.js]: It was correct earlier. This is buggy.

File: packages/web/src/components/result/ReactiveList.js [hunk 2]
```diff
@@ -176,40 +176,40 @@ class ReactiveList extends Component {
 			});
 		}
 
-		if (this.props.stream !== nextProps.stream) {
-			this.props.setStreaming(nextProps.componentId, nextProps.stream);
+		if (prevProps.stream !== this.props.stream) {
+			prevProps.setStreaming(this.props.componentId, this.props.stream);
```
Comment (metagrover) [inline packages/web/src/components/result/ReactiveList.js]: I feel you have simply replaced `this.props` -> `prevProps` and `nextProps` -> `this.props`. This won't work. Please test all the edge cases properly.

**Comment by metagrover** [comment]
Closing this PR due to incorrectness.
